src/Arrays/Programs_involving_Data_Sequences/Hard/Q1/MainTest.java

```diff
@@ -0,0 +1,65 @@
+package Arrays.Programs_involving_Data_Sequences.Hard.Q1;
+
+import global.BaseTest;
+import global.variables.*;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest extends BaseTest {
+    int[] inputs = {0, 1, 0, 2, 3, 1, 0, 2, 0};
+
+    public Clause[] testSentence() {
+        Clause[] output = new Clause[19];
+        for (int x = 0; x < 17; x += 2) {
+            output[x] = new IntegerLiteral(inputs[x / 2]);
+            output[x + 1] = new NewLine();
+        }
+        output[18] = new IntegerLiteral(3);
+        return output;
+    }
+
+    public void runMain() {
+        CountAr.main(new String[]{});
+    }
+
+    @RepeatedTest(50)
+    public void methodTestRandomNums() {
```

Comment:
After looking at this file I think the better move is to use inputProvider and hardcode a series of input arrays and the correct output counts array for them